chromadb/db/mixins/sysdb.py

```diff
@@ -776,7 +777,12 @@ def _insert_config_from_legacy_params(
         collections_t = Table("collections")
 
         # Get any existing HNSW params from the metadata
-        hnsw_metadata_params = HnswParams.extract(metadata or {})
+        metadata = metadata or {}
+        if metadata.get("hnsw:batch_size") or metadata.get("hnsw:sync_threshold"):
```

Comment:
is there any less hacky way to detect which we need?

Reply:
I think just using `PersistentHnswParams.extract()` always eliminates the need for detecting this.

Reply:
yep, discussing with @atroyn helped me realized that there's a cleaner way to do this
dbaa7c36